src/profile/v100/address/wrap_ret_address.rs

```diff
@@ -0,0 +1,150 @@
+use crate::prelude::*;
+use paste::*;
+use radix_engine_common::types::{
+    EntityType as ScryptoEntityType, NodeId as ScryptoNodeId,
+};
+use radix_engine_toolkit::models::canonical_address_types::{
+    CanonicalAccessControllerAddress as RetAccessControllerAddress,
+    CanonicalAccountAddress as RetAccountAddress,
+    CanonicalAddress as RetIsAddressTrait,
+    CanonicalAddressError as RetCanonicalAddressError,
+    CanonicalComponentAddress as RetComponentAddress,
+    CanonicalIdentityAddress as RetIdentityAddress,
+    CanonicalPackageAddress as RetPackageAddress,
+    CanonicalPoolAddress as RetPoolAddress,
+    CanonicalResourceAddress as RetResourceAddress,
+    CanonicalValidatorAddress as RetValidatorAddress,
+    CanonicalVaultAddress as RetVaultAddress,
+};
+
+pub trait AddressViaRet: Sized {
+    fn new(
+        node_id: impl Into<ScryptoNodeId>,
+        network_id: NetworkID,
+    ) -> Result<Self>;
+}
+
+macro_rules! decl_ret_wrapped_address {
```

Comment:
This macro creates the needed UniFFI converter for each RET address, using `String` as `Builtin` type. It impl `From` the wrapped RET address type into the Sargon address type. and all UniFFI exported function